File: src/ui_controller.py
#!/usr/bin/env python

from src.chef import *

class UIController:

    # Initialization creates a Chef object to which all commands will be passed
    def __init__(self):
        self.chef = Chef()

    # Returns a list of all Ingredients that MunchieMadness knows about
    def get_all_ingredients(self):
        return self.chef.get_all_ingredients()

    # Indicates that user actually has a given Ingredient
    def add_to_inventory(self, ingredient):
        self.chef.add_to_inventory(ingredient)

    # Oops, user doesn't actually have any of that Ingredient
    def remove_from_inventory(self, ingredient):
        self.chef.remove_from_inventory(ingredient)

    # Start fresh
    def clear_inventory(self):
        self.chef.clear_inventory()

    # Returns address of image corresponding to an Ingredient
    def get_image(self, ingredient):
        return self.chef.get_image(ingredient)

    # This is where the magic happens. Returns a Recipe, which has:
        # title, a string like "Peanut Butter Chocoloate Chip Popcorn Sandwich"
        # ingredients, a list of strings like ['Bread', 'Peanut Butter', 'Chocolate Chips', 'Popcorn']
        # instructions, a list of strings like ['Spread peanut butter on one or both slices of bread', 'Sprinkle with Chocolate Chips', ...]
        # TODO some sort of weirdness score?
    def generate_recipe(self):
        return self.chef.generate_recipe()

File: ui_controller.go
package main

import ()

type UIController struct {
	chef *Chef
}

// Initialization creates a Chef object to which all commands will be passed
func NewUIController() *UIController {
	return &UIController{NewChef(NewIngredientManager(), nil)}
}

// Returns a list of all Ingredients that MunchieMadness knows about
func (u *UIController) GetAllIngredients() IngredientList {
	return u.chef.GetAllIngredients()
}

// Indicates that user actually has a given Ingredient
func (u *UIController) AddToInventory(ingredient string) {
	u.chef.AddToInventory(ingredient)
}

// Oops, user doesn't actually have any of that Ingredient
func (u *UIController) RemoveFromInventory(ingredient string) {
	u.chef.RemoveFromInventory(ingredient)
}

// Start fresh
func (u *UIController) ClearInventory() {
	u.chef.ClearInventory()
}

// Returns address of image corresponding to an Ingredient
func (u *UIController) GetImage(ingredient string) string {
	return u.chef.GetImage(ingredient)
}

// This is where the magic happens. Returns a Recipe, which has:
// title, a string like "Peanut Butter Chocoloate Chip Popcorn Sandwich"
// ingredients, a list of strings like ['Bread', 'Peanut Butter', 'Chocolate Chips', 'Popcorn']
// instructions, a list of strings like ['Spread peanut butter on one or both slices of bread', 'Sprinkle with Chocolate Chips', ...]
// TODO some sort of weirdness score?
func (u *UIController) GenerateRecipe(maxScore int) *Recipe {
	return u.chef.GenerateRecipe(maxScore)
}
